### ISSUE TITLE
Unsuppressable vulnerability: BREACH attack possible in CSRF tokens

### ISSUE THREAD

**Issue by whiskeysierra** [dependency-check/DependencyCheck#1991]
**Describe the bug**
Using `org.springframework.security:spring-security-core:4.2.12.RELEASE` reports a vulnerability:

> BREACH attack possible in CSRF tokens

Suppressing it with:

```xml
<suppress>
    <gav regex="true">^org\.springframework\.security:spring-security-core:.*$</gav>
    <cpe>cpe:/a:pivotal_software:spring_security</cpe>
</suppress>
```

Changes the log message but doesn't fully suppress it:

```
[WARNING] 

One or more dependencies were identified with known vulnerabilities in Logbook: Spring Boot Auto Configuration:

spring-security-core-4.2.12.RELEASE.jar (pkg:maven/org.springframework.security/spring-security-core@4.2.12.RELEASE) : BREACH attack possible in CSRF tokens
```

The report doesn't include a `suppress` button for this.

**Version of dependency-check used**
5.0.0

**Log file**

```
[WARNING] 

One or more dependencies were identified with known vulnerabilities in Logbook: Spring Boot Auto Configuration:

spring-security-core-4.2.12.RELEASE.jar (pkg:maven/org.springframework.security/spring-security-core@4.2.12.RELEASE, cpe:2.3:a:pivotal_software:spring_security:4.2.12.release:*:*:*:*:*:*:*) : BREACH attack possible in CSRF tokens
```

https://github.com/zalando/logbook/pull/533
https://travis-ci.org/zalando/logbook/jobs/543568823

**To Reproduce**
1. Go to https://github.com/zalando/logbook/pull/533
2. Checkout branch `dependabot/maven/org.owasp-dependency-check-maven-5.0.0`
3. Run `mvn clean verify -D skipTests -P spring4`

**Expected behavior**
I want to suppress this.

**Comment by danshome** [comment]
Is there really no workaround for this?

**Comment by whiskeysierra** [comment]
I haven't found any.

**Comment by tjaronen** [comment]
I got rid of this by disabling ossindex analyzer (ossindexAnalyzerEnabled property). If I understood correctly this analyzer brings only additional info to the report.
The problem is this vulnerability https://ossindex.sonatype.org/vuln/d6ea6ac5-572d-45c3-a703-4e9726efeacd

**Comment by jeremylong** [comment]
The OSS Index Analyzer may bring in vulnerabilities that do not exist in the NVD data feeds.  With the release of 5.1.0 you can now suppress these findings using the new 1.3 suppression schema.  This ticket should have been closed on 2019-06-28 when 5.1.0 was released .- I apologize that I missed closing this one. An example of the new suppression schema is below (note, ODC is backwards compatible with previous schemas):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.3.xsd">
    <suppress>
        <notes><![CDATA[
        This suppresses a CVE identified by OSS Index using the vulnerability name and packageUrl.
        ]]></notes>
        <packageUrl regex="true">^pkg:maven/org\.eclipse\.jetty/jetty-server@.*$</packageUrl>
        <vulnerabilityName>CVE-2017-7656</vulnerabilityName>
    </suppress>
</suppressions>
```

**Comment by tjaronen** [comment]
I re-enabled OSS index Analyzer and suppressed this vulnerability now like this:
```xml 
    <suppress>
        <notes><![CDATA[
        This suppresses a CVE identified by OSS Index using the vulnerability name and packageUrl.
        ]]></notes>
        <packageUrl regex="true">^pkg:maven/org\.springframework\.security/spring-security-core@4.2.12.RELEASE$</packageUrl>
        <vulnerabilityName>BREACH attack possible in CSRF tokens</vulnerabilityName>
    </suppress>
```
